IGSIOCalibration/vtkIGSIOAbstractCalibrationAlgo.h

```diff
@@ -0,0 +1,224 @@
+  vtkSetMacro(MinimumOrientationDifferenceDeg, double);
+  vtkGetMacro(MinimumOrientationDifferenceDeg, double);
+  //@}
+
+  //@{
+  /// Required minimum amount of variation in position from the previous position in order for a transform to be accepted (0 degrees by default).
+  vtkGetMacro(PositionDifferenceThresholdMm, double);
+  vtkSetMacro(PositionDifferenceThresholdMm, double);
+  //@}
+
+  //@{
+  /// Required minimum amount of variation in position from the previous position in order for a transform to be accepted (0 degrees by default).
+  vtkGetMacro(OrientationDifferenceThresholdDegrees, double);
+  vtkSetMacro(OrientationDifferenceThresholdDegrees, double);
+  //@}
+
+  //@{
+  /// The number of input points to be stored in each bucket
+  vtkGetMacro(PoseBucketSize, int);
+  vtkSetMacro(PoseBucketSize, int);
+  //@}
+
+  //@{
+  /// The maximum number of buckets to be stored. If the number of buckets is exceeded, the oldest will be discarded.
+  vtkGetMacro(MaximumNumberOfPoseBuckets, int);
+  vtkSetMacro(MaximumNumberOfPoseBuckets, int);
+  //@}
+
+  //@{
+  /// The accepted amount of error within a single bucket. If the error in the latest bucket exceeds this amount, then all of the buckets will be discarded.
+  vtkGetMacro(MaximumPoseBucketError, double);
+  vtkSetMacro(MaximumPoseBucketError, double);
+  //@}
+
+  //@{
+  /// Output calibration tip to marker position
+  vtkGetObjectMacro(PivotPointToMarkerTransformMatrix, vtkMatrix4x4);
+  vtkSetObjectMacro(PivotPointToMarkerTransformMatrix, vtkMatrix4x4);
+  vtkGetVector3Macro(PivotPointPosition_Reference, double);
+  //@}
+
+  /// Flip the direction of the shaft axis.
+  static void FlipShaftDirection(vtkMatrix4x4* toolTipToToolMatrix);
+
+  /// Get only the rotation component from toolTipToToolMatrix.
+  static void GetToolTipToToolRotation(vtkMatrix4x4* toolTipToToolMatrix, vtkMatrix4x4* rotationMatrix);
+
+  // Helper method to compute the secondary axis, given a shaft axis
+  static vnl_vector< double > ComputeSecondaryAxis(vnl_vector< double > shaftAxis_ToolTip);
+
+protected:
+  vtkSetMacro(ErrorCode, int);
+
+  /// Get a vector containing all MarkerToReference matrices.
+  virtual std::vector<vtkMatrix4x4*> GetAllMarkerToReferenceMatrices();
+  virtual void GetAllMarkerToReferenceMatrices(std::vector<vtkMatrix4x4*>* markerToTransformMatrixArray);
+
+  /// Get a vector containing MarkerToReference in the specified bucket.
+  virtual std::vector<vtkMatrix4x4*> GetBucketMarkerToReferenceMatrices(int bucket);
+  virtual void GetBucketMarkerToReferenceMatrices(int bucket, std::vector<vtkMatrix4x4*>* matrixArray);
+
+  /// Get a vector containing MarkerToReference in the latest bucket.
+  virtual std::vector<vtkMatrix4x4*> GetLatestBucketMarkerToReferenceMatrices();
+  virtual void GetLatestBucketMarkerToReferenceMatrices(std::vector<vtkMatrix4x4*>* matrixArray);
+
+  /// Removes all data if the latest bucket is not within the acceptable error threshold.
```

Comment:
maybe `ValidateInputBuffer` would be more accurate (clean suggests to me that it just removes all content of the input buffer)